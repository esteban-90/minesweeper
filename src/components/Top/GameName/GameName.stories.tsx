import type { Story, Meta } from '@storybook/react'
import { GameName, type GameNameProps } from './GameName'

export default {
  title: 'Components/Top/GameName',
  component: GameName,
} as Meta

const Template: Story<GameNameProps> = (args) => <GameName {...args} />

export const GameNameExample = Template.bind({})

GameNameExample.args = {
  children: 'Minesweeper',
}
