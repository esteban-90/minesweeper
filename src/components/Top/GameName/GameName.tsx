import styled from '@emotion/styled'

export interface GameNameProps {
  /**
   * Text for the name
   */
  children: string
}

export const GameName = styled.h1<GameNameProps>`
  font-size: 2em;
`
