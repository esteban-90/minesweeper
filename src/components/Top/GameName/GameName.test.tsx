import { render } from '@testing-library/react'
import { GameName } from './GameName'

describe('GameName test cases:', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<GameName>Minesweeper</GameName>)
    expect(asFragment()).toMatchSnapshot()
  })
})
