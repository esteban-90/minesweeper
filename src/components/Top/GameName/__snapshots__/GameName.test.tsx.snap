// Vitest Snapshot v1

exports[`GameName test cases: > should render correctly 1`] = `
<DocumentFragment>
  <h1
    class="css-18kvaxm"
  >
    Minesweeper
  </h1>
</DocumentFragment>
`;
